#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/Floor.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/kuka/youbot/protos/Youbot.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/kuka/youbot/protos/KukaBox.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/apartment_structure/protos/Wall.proto"

WorldInfo {
  info [
    "Simulation of the KUKA youBot robot."
    "This model includes an accurate and fast simulation of the Meccanum wheels."
    "The robot can be equiped with various sensors (including the Microsoft Kinect) and configured with zero, one or two arms."
  ]
  title "KUKA youBot"
  basicTimeStep 16
  contactProperties [
    ContactProperties {
      material1 "InteriorWheelMat"
      coulombFriction [
        0, 2, 0
      ]
      frictionRotation -0.785398 0
      bounce 0
      forceDependentSlip [
        10, 0
      ]
    }
    ContactProperties {
      material1 "ExteriorWheelMat"
      coulombFriction [
        0, 2, 0
      ]
      frictionRotation 0.785398 0
      bounce 0
      forceDependentSlip [
        10, 0
      ]
    }
  ]
}
Viewpoint {
  orientation -0.35099234520089523 0.29889182642700385 0.8873939653307912 1.8473591078909313
  position 2.4389137819701667 -10.142467412995096 7.83573141586973
  follow "youBot"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
  appearance Appearance {
    material Material {
      ambientIntensity 1
      emissiveColor 1 1 1
      shininess 0.8
      specularColor 1 1 1
    }
    texture ImageTexture {
      url [
        "C:/Users/DELL/Downloads/Telegram Desktop/lastdesign100.jpg"
      ]
      repeatS FALSE
      repeatT FALSE
      filtering 5
    }
    textureTransform TextureTransform {
      scale 0.05 0.05
    }
  }
}
Wall {
  translation 0.89 0 -0.16
  size 0.1 0.3 0.2
}
Wall {
  translation 0.98 0.109998 -0.18
  rotation 0 0 1 1.5708
  name "wall(1)"
  size 0.1 0.3 0.2
}
Wall {
  translation 0.98 -0.050002 -0.18
  rotation 0 0 1 1.5708
  name "wall(2)"
  size 0.1 0.3 0.2
}
Wall {
  translation 1.05 0.00999838 -0.16
  rotation 0 0 1 5.307156870259904e-06
  name "wall(3)"
  size 0.1 0.3 0.2
}
Transform {
  scale 0.5 0.5 0.5
  children [
    Pose {
    }
  ]
}
Shape {
}
Transform {
  scale 0.1 0.1 0.1
}
Youbot {
  hidden position_0_0 1.7845955187816305
  hidden position_0_1 1.7845955260747965
  hidden position_0_2 1.7845937618753178
  hidden position_0_3 1.7845937680258672
  hidden linearVelocity_0 -0.7043495825727027 0.001069395766374988 -0.0028243433442212505
  hidden angularVelocity_0 -0.00014815697944770343 -0.05886796485420699 1.121242258224781e-06
  hidden position_15_0 -1.3105551767389374e-06
  hidden rotation_16 0 0 -1 1.310624684530283e-06
  hidden position_16_0 0.0004055828371061358
  hidden linearVelocity_16 -0.7144842079605925 0.001094964509144375 -0.012274442139914906
  hidden angularVelocity_16 -0.00014813525946812649 -0.058910046371891615 -1.8841546060806725e-05
  hidden rotation_17 0 -1 0 0.00040558283687667693
  hidden position_17_0 8.376959416509405e-07
  hidden linearVelocity_17 -0.7207581280672792 0.001110677360912161 -0.013371266770631787
  hidden angularVelocity_17 -0.00015571265772449356 -0.06374103713225236 -1.8276774444684603e-05
  hidden rotation_18 0 -0.9999999999999999 0 8.376520468327582e-07
  hidden position_18_0 1.9439305724947127e-07
  hidden linearVelocity_18 -0.7298927996626314 0.0011329683704180468 -0.013374119393876878
  hidden angularVelocity_18 -0.0001553922602171438 -0.06375216715709076 -1.906302707042828e-05
  hidden rotation_19 0 -1 0 1.954268969387413e-07
  hidden position_19_0 4.1603641788006697e-08
  hidden linearVelocity_19 -0.7360594660411391 0.0011480038341315096 -0.013375447297299147
  hidden angularVelocity_19 -0.00015539705068124814 -0.06375478825700837 -1.9065628023618862e-05
  hidden rotation_20 0 0 1 4.2146848510894035e-08
  hidden position_20_0 2.3931410079208083e-09
  hidden position_20_1 -2.4087412035644107e-09
  hidden linearVelocity_20 -0.7424476884414822 0.0011636014677593347 -0.013360467529373215
  hidden angularVelocity_20 -0.0001553961510409279 -0.0637555279432977 -1.853194494874477e-05
  hidden translation_21 0 2.3931410079208083e-09 0.09
  hidden linearVelocity_21 -0.7465932515141572 0.0011737278884034696 -0.013347358992969663
  hidden angularVelocity_21 -0.00015539379382345476 -0.06375556888360694 -1.8538250067710697e-05
  hidden translation_22 0 2.4087412035644107e-09 0.09
  hidden linearVelocity_22 -0.746592837521588 0.0011737275362470447 -0.01334853106710394
  hidden angularVelocity_22 -0.00015539865635801487 -0.06375556887547863 -1.85256370648537e-05
  hidden position_23_0 3.352521892051834e-06
  hidden rotation_24 0 0 1 3.352463232145584e-06
  hidden position_24_0 -0.0004151302251185067
  hidden linearVelocity_24 -0.7144265479913352 0.0010953573211374518 0.00789176139736507
  hidden angularVelocity_24 -0.00014816351621982365 -0.05890623715460796 4.0774304552444176e-05
  hidden rotation_25 0 1 0 0.0004151302253051622
  hidden position_25_0 -8.393100796568603e-07
  hidden linearVelocity_25 -0.7207020871891708 0.0011113898140773413 0.008940404383856498
  hidden angularVelocity_25 -0.00015555289362735773 -0.06385087640695856 4.050724211327295e-05
  hidden rotation_26 0 1 0 8.392410183148877e-07
  hidden position_26_0 -1.946148717735241e-07
  hidden linearVelocity_26 -0.7298532384681008 0.001133686631035339 0.00894448956597588
  hidden angularVelocity_26 -0.0001557798605429219 -0.0638620218203525 4.129396297562242e-05
  hidden rotation_27 0 1 0 1.954268969387413e-07
  hidden position_27_0 -4.1654972686164585e-08
  hidden linearVelocity_27 -0.7360300171074151 0.0011487590894102232 0.00895314014531369
  hidden angularVelocity_27 -0.00015578328690851056 -0.06386464462738911 4.12966247414693e-05
  hidden rotation_28 0 0 -0.9999999999999999 4.2146848510894035e-08
  hidden position_28_0 -2.3300598973109972e-09
  hidden position_28_1 2.3132412279445295e-09
  hidden linearVelocity_28 -0.7424292182606403 0.001164414300274517 0.008970108804880066
  hidden angularVelocity_28 -0.00015578651187734895 -0.06386538490838313 4.0762366270995427e-05
  hidden translation_29 0 -2.3300598973109972e-09 0.09
  hidden linearVelocity_29 -0.746582101083195 0.0011745822602497332 0.00898228721650305
  hidden angularVelocity_29 -0.00015578808068392823 -0.06386542587302886 4.076867485284337e-05
  hidden translation_30 0 -2.3132412279445295e-09 0.09
  hidden linearVelocity_30 -0.7465813283361412 0.001174580849050261 0.008983459650545957
  hidden angularVelocity_30 -0.0001557850511763934 -0.0638654258803503 4.0756059815878244e-05
  hidden rotation_31 0 1 0 1.7845955187816305
  hidden linearVelocity_31 -0.7033113170744664 0.0013432357536240398 -0.015196436143321319
  hidden angularVelocity_31 -0.021323082815878767 -14.058856563009984 0.0005474820122867673
  hidden rotation_32 0 1 0 1.7845955260747965
  hidden linearVelocity_32 -0.7033126427044546 0.0007905806584693129 -0.015178243418154855
  hidden angularVelocity_32 -0.02133371221247061 -14.058856565037505 0.0005271848997049732
  hidden rotation_33 0 1 0 1.7845937618753178
  hidden linearVelocity_33 -0.703190836336458 0.0007457833153401282 0.011123816249166146
  hidden angularVelocity_33 -0.021332173171943167 -14.058848267814108 0.000520407653866912
  hidden rotation_34 0 1 0 1.7845937680258672
  hidden linearVelocity_34 -0.7031903317117928 0.0013879771178259797 0.011142008976557852
  hidden angularVelocity_34 -0.02132467226441508 -14.058848292930122 0.0005542616589301098
  translation 2.8436613325971623 -2.578002069088875 0.10135606929865032
  rotation 0.0013863224059678793 1.6997849112783043e-05 0.9999990389101681 3.140080000710372
  controller "my_controller"
  controllerArgs [
    "demo"
  ]
  supervisor TRUE
  bodySlot [
    Emitter {
      channel 1
    }
    Camera {
      translation 0.285 -0.01 -0.0959809
      rotation 0 -1 0 4.6928012867022945e-06
      name "front-camera"
      physics Physics {
      }
      fieldOfView 0.05
    }
    DistanceSensor {
      translation 0 -0.0115 0.007
      rotation 0 1 0 1.57
      name "lfs0"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 -0.0085 0.007
      rotation 0 1 0 1.57
      name "lfs1"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 -0.0065 0.007
      rotation 0 1 0 1.57
      name "lfs2"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 0 0.007
      rotation 0 1 0 1.57
      name "lfs3"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 0 0.007
      rotation 0 1 0 1.57
      name "lfs4"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 0.0065 0.007
      rotation 0 1 0 1.57
      name "lfs5"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 0.0085 0.007
      rotation 0 1 0 1.57
      name "lfs6"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 0.0115 0.007
      rotation 0 1 0 1.57
      name "lfs7"
      type "infra-red"
    }
    GPS {
    }
    InertialUnit {
    }
    Camera {
      translation 0.050074 -0.01 -0.0807565
      rotation 0 1 0 1.57
    }
    DistanceSensor {
      translation 0.29 0 0
      rotation 1 0 0 1.57
      lookupTable [
        0 0 0
        10 10000 0
      ]
    }
  ]
  numberOfArms 2
}
KukaBox {
  hidden linearVelocity_0 2.7906532220296607e-18 -3.048570206438373e-18 0.016724149566566412
  hidden angularVelocity_0 -3.877913463250065e-16 6.964609996491867e-18 7.182210266288436e-17
  translation -0.413383 -2.10937 0.02386135015058442
  rotation -1.564597824611814e-17 1.5598668087679862e-17 0.9999999999999999 3.14
  name "yellow"
  color 1 1 0
  size 0.05 0.05 0.05
  contactMaterial "box"
  physics Physics {
    density 2700
    centerOfMass [
      0 0 0
    ]
    damping Damping {
    }
  }
}
KukaBox {
  hidden linearVelocity_0 -2.787711443973124e-17 -7.35575186543067e-17 0.016724149566566426
  hidden angularVelocity_0 2.7574555872704273e-15 -1.1814876811410013e-15 1.1462109108717455e-16
  translation -0.945425 -3.38848 0.02386135015058442
  rotation -6.163127460088198e-18 -1.2948293677091197e-17 0.9999999999999999 3.14
  name "blue"
  color 0 0 1
  size 0.05 0.05 0.05
  contactMaterial "box"
  physics Physics {
    density 2700
    centerOfMass [
      0 0 0
    ]
    damping Damping {
    }
  }
}
KukaBox {
  hidden linearVelocity_0 1.41552028642324e-18 -1.8722995328360624e-17 0.016724149566566416
  hidden angularVelocity_0 1.1565676363036372e-15 4.001354552192016e-16 4.056555573459954e-17
  translation -0.942456 -1.52849 0.023861350150584423
  rotation -1.204839600487907e-17 -6.212140309897642e-17 0.9999999999999999 3.14
  name "green"
  color 0 1 0
  size 0.05 0.05 0.05
  contactMaterial "box"
  physics Physics {
    density 2700
    centerOfMass [
      0 0 0
    ]
    damping Damping {
    }
  }
}
KukaBox {
  hidden linearVelocity_0 -1.755908023074441e-17 8.700340939861369e-18 0.016724149566566444
  hidden angularVelocity_0 -3.9876811942182073e-16 -1.1646375219443155e-15 2.1150434997478043e-17
  translation -2.38437 -2.73603 0.02386135015058442
  rotation -4.399000686663221e-17 -9.063617762647551e-18 0.9999999999999999 3.14
  name "red"
  color 1 0 0
  size 0.05 0.05 0.05
  contactMaterial "box"
  physics Physics {
    density 2700
    centerOfMass [
      0 0 0
    ]
    damping Damping {
    }
  }
}
DEF Carpet Pose {
  translation 2.07999 -2.57 -0.0624748
  rotation 6.17457808927593e-07 0.7071077811853963 0.7071057811860152 3.14159
  children [
    Transform {
      translation -0.4 0.060000000000000005 0
      scale 0.5 0.5 0.5
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 1 1 0
            }
          }
          geometry Box {
            size 0.2 0.01 0.2
          }
        }
      ]
    }
    Transform {
      translation -0.2 0.060000000000000005 0
      scale 0.5 0.5 0.5
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 1 0
            }
          }
          geometry Box {
            size 0.2 0.01 0.2
          }
        }
      ]
    }
    Transform {
      translation 0 0.060000000000000005 0
      scale 0.5 0.5 0.5
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 1
            }
          }
          geometry Box {
            size 0.2 0.01 0.2
          }
        }
      ]
    }
    Transform {
      translation 0.20000000000000007 0.060000000000000005 0
      scale 0.5 0.5 0.5
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 1 0 0
            }
          }
          geometry Box {
            size 0.2 0.01 0.2
          }
        }
      ]
    }
  ]
}
Youbot {
  hidden position_0_0 -1.9164662046280976e-09
  hidden position_0_1 -1.4792904678984087e-09
  hidden position_0_2 -6.647115404906367e-10
  hidden position_0_3 2.0770608077151422e-09
  hidden linearVelocity_0 -6.440033617034984e-05 5.258142666557038e-06 -0.0022261596539569477
  hidden angularVelocity_0 -7.974283442902639e-05 -0.0009787300607028092 -9.833849496194e-09
  hidden position_16_0 -1.0323048587655618e-06
  hidden rotation_17 0 0 -1 1.0323827311807255e-06
  hidden position_17_0 9.677620140790098e-08
  hidden linearVelocity_17 -0.00022768001300721766 1.8696824775118487e-05 -0.002456602373108921
  hidden angularVelocity_17 -7.960743972776528e-05 -0.0009806756964553193 -1.0012130957517249e-05
  hidden rotation_18 0 -1 0 9.884312124119402e-08
  hidden position_18_0 1.477653824837956e-07
  hidden linearVelocity_18 -0.00032337361842627716 2.670126672793397e-05 -0.0025206694863547634
  hidden angularVelocity_18 -7.978941241716293e-05 -0.0009798511600705331 -1.0011907838915093e-05
  hidden rotation_19 0 -1 0 1.490116119384766e-07
  hidden position_19_0 -7.621972576263293e-10
  hidden linearVelocity_19 -0.00046299714478048506 3.8022329218972284e-05 -0.002544640824779532
  hidden angularVelocity_19 -7.951144525905707e-05 -0.0009784556550551222 -1.0012582913475728e-05
  hidden position_20_0 -2.2239337584904417e-10
  hidden linearVelocity_20 -0.0005567325724436376 4.566005655070149e-05 -0.0025621200786452064
  hidden angularVelocity_20 -7.951183057225044e-05 -0.0009784610639579677 -1.001257636337611e-05
  hidden position_21_0 -3.890124444546052e-11
  hidden position_21_1 1.3237368540516813e-11
  hidden linearVelocity_21 -0.0006545875724229421 5.361311808086516e-05 -0.002566277796471794
  hidden angularVelocity_21 -7.951193778649695e-05 -0.000978461922873103 -1.0014656896264983e-05
  hidden translation_22 0 -3.890124444546052e-11 0.09
  hidden linearVelocity_22 -0.0007182891455773559 5.8781936704909945e-05 -0.002565929724838658
  hidden angularVelocity_22 -7.950996504968158e-05 -0.0009784619742128737 -1.0014663189720143e-05
  hidden linearVelocity_23 -0.0007180880760190096 5.878197128871877e-05 -0.0025674888171595663
  hidden angularVelocity_23 -7.951392278779257e-05 -0.0009784619682039726 -1.0014650526445235e-05
  hidden position_24_0 1.0118559395471796e-06
  hidden rotation_25 0 0 1 1.0119632986992495e-06
  hidden position_25_0 -9.563077186039646e-08
  hidden linearVelocity_25 -0.0002276730980373172 1.871905367215203e-05 -0.002123313143941975
  hidden angularVelocity_25 -7.988759305976164e-05 -0.0009768688467893618 9.7975940807004e-06
  hidden rotation_26 0 0.9999999999999999 0 9.424321830774485e-08
  hidden position_26_0 -1.4810438168644293e-07
  hidden linearVelocity_26 -0.00032302582040556033 2.671997061430183e-05 -0.0021458342213661087
  hidden angularVelocity_26 -7.97100444590586e-05 -0.0009760539049505671 9.797311007501019e-06
  hidden rotation_27 0 1 0 1.490116119384766e-07
  hidden position_27_0 7.596985854314276e-10
  hidden linearVelocity_27 -0.00046209470145616726 3.8064077576025715e-05 -0.002177982999566505
  hidden angularVelocity_27 -7.998623052043628e-05 -0.0009746546039237534 9.797887418034777e-06
  hidden position_28_0 2.2171045585906684e-10
  hidden linearVelocity_28 -0.0005554718979115279 4.5746611059542564e-05 -0.002191925084202893
  hidden angularVelocity_28 -7.9986617070354e-05 -0.0009746600032646655 9.79788110755287e-06
  hidden position_29_0 1.3457881455214679e-11
  hidden position_29_1 -3.905186702112677e-11
  hidden linearVelocity_29 -0.0006529468690818066 5.3746181850062674e-05 -0.002196077635133578
  hidden angularVelocity_29 -7.998672664806552e-05 -0.000974660861183407 9.799954563005777e-06
  hidden linearVelocity_30 -0.0007163984969805039 5.894557993785398e-05 -0.0021972926842687536
  hidden angularVelocity_30 -7.998870937468973e-05 -0.0009746609061826973 9.799960854588249e-06
  hidden translation_31 0 3.905186702112677e-11 0.09
  hidden linearVelocity_31 -0.0007162032738016032 5.894491411737066e-05 -0.00219572403002631
  hidden angularVelocity_31 -7.998475620899551e-05 -0.0009746609127726227 9.799948193917133e-06
  hidden linearVelocity_32 -4.918939429884112e-05 4.064288083507643e-06 -0.0023268513362777535
  hidden angularVelocity_32 -7.973964303200978e-05 -0.0009787389504379415 -9.84511856687917e-09
  hidden linearVelocity_33 -4.9185509532648046e-05 3.966131663433339e-06 -0.002302628786861831
  hidden angularVelocity_33 -7.974459912099008e-05 -0.000978741504580509 -9.851713926314583e-09
  hidden linearVelocity_34 -4.939482065936233e-05 3.95706213416433e-06 -0.0018892382692533981
  hidden angularVelocity_34 -7.974575342571593e-05 -0.0009787514696719965 -9.850518962017682e-09
  hidden linearVelocity_35 -4.942906508544155e-05 4.104685510169171e-06 -0.0018650157198377552
  hidden angularVelocity_35 -7.973841986978224e-05 -0.0009787489103683295 -9.845599466810513e-09
  translation 2.8092966828712567 2.7453738960069254 0.10136063917887447
  rotation -0.0002537837109742076 1.8342236058958904e-05 0.9999999676286947 3.1399999985606857
  name "SecBot"
  controller "SecController"
  controllerArgs [
    "demo"
  ]
  supervisor TRUE
  bodySlot [
    Receiver {
      channel 1
      allowedChannels [
        1
      ]
    }
    Emitter {
      channel 1
      allowedChannels [
        1
      ]
    }
    Camera {
      translation 0.285 -0.01 -0.0559809
      rotation 0 -1 0 4.6928012867022945e-06
      name "front-camera"
      physics Physics {
      }
      fieldOfView 0.05
    }
    DistanceSensor {
      translation 0 -0.0115 0.007
      rotation 0 1 0 1.57
      name "lfs0"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 -0.0085 0.007
      rotation 0 1 0 1.57
      name "lfs1"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 -0.0065 0.007
      rotation 0 1 0 1.57
      name "lfs2"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 0 0.007
      rotation 0 1 0 1.57
      name "lfs3"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 0 0.007
      rotation 0 1 0 1.57
      name "lfs4"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 0.0065 0.007
      rotation 0 1 0 1.57
      name "lfs5"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 0.0085 0.007
      rotation 0 1 0 1.57
      name "lfs6"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 0.0115 0.007
      rotation 0 1 0 1.57
      name "lfs7"
      type "infra-red"
    }
    GPS {
    }
    InertialUnit {
    }
    Camera {
      translation 0.050074 -0.01 -0.0807565
      rotation 0 1 0 1.57
    }
    DistanceSensor {
      translation 0.29 0 0
      rotation 1 0 0 1.57
      lookupTable [
        0 0 0
        10 10000 0
      ]
    }
  ]
  numberOfArms 2
}
